src/org/omegat/core/segmentation/MapRule.java

```diff
@@ -44,56 +46,99 @@
+     */
     public MapRule(String language, String pattern, List<Rule> rules) {
-        this.setLanguage(language);
+        String code = LanguageCodes.getLanguageCodeByPattern(pattern);
+        this.setLanguage(code != null ? code : language);
         this.setPattern(pattern);
         this.setRules(rules);
     }
 
-    /** Language Name */
-    private String languageCode;
-
-    public MapRule(Languagemap languagemap, List<Rule> rules) {
-        this.setLanguage(languagemap.getLanguagerulename());
-        this.setPattern(languagemap.getLanguagepattern());
-        this.setRules(rules);
-    }
-
     /** Returns Language Name (to display it in a dialog). */
     public String getLanguageName() {
+        /*
+         * When there has already migrated a SRX file store, languageCode fields
+         * has a name defined as "LanguageCodes.*_CODE". Otherwise, MapRule
+         * object is created from "segmentation.conf" java beans file, so it is
+         * localized name of language. We first assume the latter. If res is
+         * empty, the object is created from a SRX file, then return
+         * languageCode itself.
+         */
         String res = LanguageCodes.getLanguageName(languageCode);
         return StringUtil.isEmpty(res) ? languageCode : res;
     }
 
     /** Sets Language Code */
     public void setLanguage(String code) {
+        /*
+         * setLanguage method is called from XmlDecoder of a Java beans library
+         * when migrating from "segmentation.conf" beans file. An argument will
+         * be localized name of language. When the object is created from a
+         * standard SRX file, the argument will be standard language name,
+         * defined as "LanguageCodes.*_CODE". The behavior was changed in OmegaT
+         * 6.0.0 release in 2023. We first detect whether the argument is
+         * standard code. If the code is not a standard code, then try to find a
+         * localized name of the language name. When you read the comment long
+         * after OmegaT 6.x, and you believe all the OmegaT 4.x and 5.x users
+         * are migrated to OmegaT 6.x or later, you may want to remove the chunk
+         * below.
+         */
         if (!LanguageCodes.isLanguageCodeKnown(code)) {
             String alt = LanguageCodes.getLanguageCodeByName(code);
             if (alt != null) {
                 languageCode = alt;
-                return;
             } else {
-                Log.logWarningRB("CORE_SRX_RULES_UNKNOWN_LANGUAGE_CODE", code);
+                // migration heuristics: Germany translation changed in v5.5.
+                // See:
+                // https://github.com/omegat-org/omegat/pull/1158#issuecomment-2448788253
+                if (code != null && code.contains("Textdateien")) {
+                    languageCode = LanguageCodes.F_TEXT_CODE;
+                } else {
+                    LOGGER.atDebug().setMessageRB("CORE_SRX_RULES_UNKNOWN_LANGUAGE_CODE").addArgument(code)
```

Comment:
```suggestion
                    LOGGER.atDebug().setMessage("Unknown language code '{}' specified").addArgument(code)
```
Debug message does not use localized message.